src/components/asset-list/RecyclerAssetList2/FastComponents/FastCoinIcon.tsx

```diff
@@ -0,0 +1,167 @@
+import React, { useCallback, useState } from 'react';
+import { Image, StyleSheet, View } from 'react-native';
+import CoinIcon from '../../../coin-icon/CoinIcon';
+import { FastChainBadge } from './FastCoinBadge';
+import EthIcon from '@rainbow-me/assets/eth-icon.png';
+import { AssetType } from '@rainbow-me/entities';
+import { useColorForAsset } from '@rainbow-me/hooks';
+import { ImageWithCachedMetadata, ImgixImage } from '@rainbow-me/images';
+import { borders, fonts } from '@rainbow-me/styles';
+import {
+  FallbackIcon,
+  getTokenMetadata,
+  getUrlForTrustIconFallback,
+  isETH,
+} from '@rainbow-me/utils';
+
+const fallbackTextStyles = {
+  fontFamily: fonts.family.SFProRounded,
+  fontWeight: fonts.weight.bold,
+  letterSpacing: fonts.letterSpacing.roundedTight,
+  marginBottom: 0.5,
+  textAlign: 'center',
+};
+
+const fallbackIconStyle = {
+  ...borders.buildCircleAsObject(40),
+  position: 'absolute',
+};
+
+const imagesCache: { [imageUrl: string]: boolean } = {};
+
+const CoinIconWithBackground = React.memo(function CoinIconWithBackground({
+  imageUrl,
+  symbol,
+  shadowColor,
+}: {
+  imageUrl: string;
+  symbol: string;
+  shadowColor: string;
+}) {
+  const key = `${symbol}-${imageUrl}`;
+
+  const isCached = imagesCache[key];
+
+  // this is a hack
+  // we should default to trying to render the image component to fetch the image
+  // then we cache the result - is the image available or not
+  // and then we default to the result
+  const shouldShowImage = typeof isCached === 'undefined' ? true : isCached;
+  const [, forceRerender] = useState(0);
+
+  const onLoad = useCallback(() => {
+    imagesCache[key] = true;
```

Comment:
like in those places do `setIsCached(imagesCache[key])`
and then 
`const [isCached, setIsCached] = useState(imagesCache[key])`

Reply:
This whole thing is gonna be gone when we merge your react-coin-icon PNGs PR.

I added explanation in the comments

Reply:
"we should default to trying to render the image component to fetch the image". What's the alternative? 
I don't really understand this explanation. If you want to have a cache outside of the component, can we use recoil that can be static and reactive? 

Now we have this function if this might help https://github.com/rainbow-me/rainbow/blob/develop/patches/react-native-fast-image%2B8.5.11.patch#L153

Reply:
Can you try to get rid of the force rendering and just use the local state initially populated by the object? In not can you show me what's not working? I don't think this is a good thing to have in the codebase.

Reply:
Ok, will try it too